### PR TITLE
Android audio fixes

### PR DESCRIPTION
**Pull request by gcalero**
-- This PR replaces 13589 --

The audio in android is quiet and very choppy. We've detected many issues:
- In samsung devices the audio input stream is very low volume
- In some android devices the input device suddenly stops bringing input data. There is a workaround that checks the amount of input reads in the last 2 seconds and if this amount is less than a threshold it stops+starts the input device. A bug here was causing unwanted (and audible) input restarts.
- Some QObject::connect may be causing the audio input to start (and work) in the main thread. This PR fixes that.

Test Plan
- Make a conversation between an android client and a desktop client. Desktop should be able to hear an understand everything what is said in android.
- An easy way to listen for audio dropouts, is to play a pure sine wave i.e. 
 ```injector = Audio.playSound(SoundCache.getSound("http://xiph.org/~xiphmont/demo/1kHz-0-dither.wav"), { position: MyAvatar.position, loop: false, localOnly: false, volume: 1.0 } );``` 

In a more primitive way you can play the sound near to the mic and check it at the other end (i.e. desktop client)
- If possible, detect any difference in audio quality on different domains (i.e heavy loaded domain vs. almost empty domain)
- The audio quality should be better than before this PR but it's useful to know conditions in which it fails
- For the same conditions, android audio should perform similarly to desktop audio **connected to wifi**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11404/HighFidelity-Beta-PR13636-c8a168c.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11404/HighFidelity-Beta-PR13636-c8a168c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11404/HighFidelity-Beta-Interface-PR13636-c8a168c.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11418/HighFidelity-Beta-PR13636-d230a63.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11418/HighFidelity-Beta-PR13636-d230a63.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11418/HighFidelity-Beta-Interface-PR13636-d230a63.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11418/HighFidelity-Beta-PR13636-d230a63.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11418/HighFidelity-Beta-Interface-PR13636-d230a63.exe

File: libraries/audio-client/src/AudioClient.cpp [hunk 1]
```diff
@@ -482,6 +486,15 @@ bool nativeFormatForAudioDevice(const QAudioDeviceInfo& audioDevice,
     audioFormat.setSampleType(QAudioFormat::SignedInt);
     audioFormat.setByteOrder(QAudioFormat::LittleEndian);
 
+#if defined(Q_OS_ANDROID)
+    // Using the HW sample rate (AUDIO_INPUT_FLAG_FAST) in some samsung phones causes a low volume at input stream
+    // Changing the sample rate forces a resampling that (in samsung) amplifies +18 dB
+    QAndroidJniObject brand =  QAndroidJniObject::getStaticObjectField<jstring>("android/os/Build", "BRAND");
+    if (audioDevice == QAudioDeviceInfo::defaultInputDevice() && brand.toString().contains("samsung", Qt::CaseInsensitive)) {
```
Comment (kencooke) [inline libraries/audio-client/src/AudioClient.cpp]: It's worth testing if forcing both input and output to 24000 (and letting Android do all resampling) works better or worse than input only. But it's not necessary for this PR.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11433/HighFidelity-Beta-PR13636-988acdb.apk

File: libraries/audio-client/src/AudioClient.cpp (hunk 1 above)
Comment (gcalero) [reply libraries/audio-client/src/AudioClient.cpp]: This build has both input and output sample rate at 24khz:  https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11433/HighFidelity-Beta-PR13636-988acdb.apk. Reverting now

Comment (gcalero) [reply libraries/audio-client/src/AudioClient.cpp]: I don't see any improvement testing in a local network

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11434/HighFidelity-Beta-PR13636-00e3de4.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11434/HighFidelity-Beta-PR13636-00e3de4.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11434/HighFidelity-Beta-Interface-PR13636-00e3de4.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11434/HighFidelity-Beta-PR13636-00e3de4.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11434/HighFidelity-Beta-Interface-PR13636-00e3de4.exe

File: libraries/audio-client/src/AudioClient.cpp (hunk 1 above)
Comment (kencooke) [reply libraries/audio-client/src/AudioClient.cpp]: Ok, thanks for testing both ways.

Comment (gcalero) [reply libraries/audio-client/src/AudioClient.cpp]: no problem

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Fail @qliacoombs @Qliemillar 
Audio levels for Android to Desktop and Android to Android are still choppy and drop out.
Logs and sample video attached.
[Android Sound.zip](https://github.com/highfidelity/hifi/files/2215557/Android.Sound.zip)

**Comment by divyaelango88** [comment]
build this please

**Comment by divyaelango88** [comment]
@qlimnavarro what domain did you guys do the test on? 
Is the experience better with the PR than without it?

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11472/HighFidelity-Beta-PR13636-00e3de4.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11472/HighFidelity-Beta-PR13636-00e3de4.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11472/HighFidelity-Beta-Interface-PR13636-00e3de4.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11472/HighFidelity-Beta-PR13636-00e3de4.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11472/HighFidelity-Beta-Interface-PR13636-00e3de4.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11488/HighFidelity-Beta-PR13636-3c49e77.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11488/HighFidelity-Beta-PR13636-3c49e77.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11488/HighFidelity-Beta-Interface-PR13636-3c49e77.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11488/HighFidelity-Beta-PR13636-3c49e77.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11488/HighFidelity-Beta-Interface-PR13636-3c49e77.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11520/HighFidelity-Beta-PR13636-f24cfeb.apk

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11520/HighFidelity-Beta-PR13636-f24cfeb.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11520/HighFidelity-Beta-Interface-PR13636-f24cfeb.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11520/HighFidelity-Beta-PR13636-f24cfeb.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11520/HighFidelity-Beta-Interface-PR13636-f24cfeb.exe

**Comment by qlimnavarro** [comment]
Pass @conklin94122 @divyaelango88 
Audio tests passed with dev approval.

**Comment by qlilaseguinolaza** [comment]
Currently testing

**Pull request merged**